### DISCUSSION TITLE
Is it possible to use a joystick as a midi controller for navigating the redbox in ableton?

### DISCUSSION THREAD

**Discussion by bramlauwers** [tttapa/Control-Surface#1016] [General]
Hi,

Thank you for your great library. I'am looking for a way to use a joystick to control the redbox in ableton. So when i flick the joystick down, the redbox goes down, when i flick it to the left, it goes to the left.

I know how to get this into a script for ableton. But I can't get the joystick code to work and give me the CC messages.

Is there existing code? 

Thank you

**Comment by tttapa** [comment]
There's no existing code for this, but it shouldn't be too hard to do. If you just have an analog joystick, you could use the FilteredAnalog class to read the X and Y values, and using some if-statements with appropriate thresholds, you can get a basic implementation working quite easily. If you want more advanced features such as moving while the joystick is being held to one side, or moving faster the farther you move the joystick, this will require a bit more code, but nothing too complicated.

Then you'll also need to figure out which MIDI messages you need to send for Ableton to move the box around. This is not something I can help you with, you'll have to dig into the Ableton documentation. A quick Google search reveals some discussions about it on https://forum.ableton.com/viewtopic.php?f=4&t=183459, but I have never tried anything like it myself.

**Reply by bramlauwers** [reply]
Thank you for the respond. I'am quite new to arduino and learning on the go. I've written some code wich serialprint the right direction. 
Now i'am looking for the right code to send a CC message. I was looking into the "sendCC()," but can't get it to work. Could you please help me? 
 

```cpp
#include <Control_Surface.h>
USBMIDI_Interface midi;

void setup() {

  Serial.begin(9600);
}

void loop() {
  
  int PosX = analogRead(27);
  int PosY = analogRead(26);
  Serial.println(PosX);
  Serial.print("\t");
  Serial.println(PosY);
  delay(1);  

if ((PosX > 1000) && (PosY > 800 ) && (PosY < 1000)) {Serial.println("down");}
else {}

if (PosX <10) {Serial.println("up");}
else {}

if ((PosX > 800) && (PosY > 1000 ) && (PosX < 1000)) {Serial.println("left");}
else {}

if (PosY <10) {Serial.println("right");}
else {}

}
```

**Reply by tttapa** [reply]
For examples of sending MIDI messages, have a look at https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/df7/midi-tutorial.html#midi_md-sending and https://tttapa.github.io/Control-Surface-doc/Doxygen/d5/d35/Send-All-MIDI-Messages_8ino-example.html.
